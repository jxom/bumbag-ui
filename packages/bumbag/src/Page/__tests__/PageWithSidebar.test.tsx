import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { PageWithSidebar } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<PageWithSidebar sidebar={<div>this is a sidebar</div>}>Hello world</PageWithSidebar>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with sidebarPlacement', () => {
    const { container } = render(
      <PageWithSidebar sidebar={<div>this is a sidebar</div>} sidebarPlacement="right">
        Hello world
      </PageWithSidebar>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <PageWithSidebar sidebar={<div>this is a sidebar</div>} color="primary">
        Hello world
      </PageWithSidebar>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('variants', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(
      <PageWithSidebar sidebar={<div>this is a sidebar</div>} variant="test">
        hello world
      </PageWithSidebar>,
      {
        theme: {
          PageWithSidebar: {
            variants: { test: { styles: { base: { backgroundColor: 'red' } } } },
          },
        },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(
      <PageWithSidebar
        sidebar={<div>this is a sidebar</div>}
        overrides={{
          PageWithSidebar: { styles: { base: { backgroundColor: 'red' } } },
        }}
      >
        hello world
      </PageWithSidebar>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(
      <PageWithSidebar sidebar={<div>this is a sidebar</div>}>hello world</PageWithSidebar>,
      {
        theme: {
          PageWithSidebar: { styles: { base: { backgroundColor: 'red' } } },
        },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(
      <PageWithSidebar sidebar={<div>this is a sidebar</div>}>hello world</PageWithSidebar>,
      {
        theme: {
          PageWithSidebar: { defaultProps: { className: 'test', color: 'primary' } },
        },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
