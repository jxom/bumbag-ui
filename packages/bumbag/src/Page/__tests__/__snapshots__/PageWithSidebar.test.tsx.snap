// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-14 {
  min-height: 100vh;
}

.emotion-4 {
  background-color: var(--bb-palette-background,#000000);
  height: 100vh;
  min-width: 320px;
  width: 320px;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  border-right: 1px solid var(--bb-palette-white800,#000000);
}

.emotion-7 {
  position: fixed;
  z-index: 999999;
  overflow-y: scroll;
}

@media screen and (max-width:1023px) {
  .emotion-7 {
    display: none;
  }
}

.emotion-13 {
  width: 100%;
  padding-left: 320px;
  padding-left: 0px;
}

@media screen and (max-width:1023px) {
  .emotion-13 {
    padding-left: 0px;
  }
}

.emotion-16 {
  color: var(--bb-palette-primary,#000000) !important;
}

<div
  class="emotion-14 emotion-0 emotion-16 test bb-Box bb-PageWithSidebar"
>
  <div
    class="emotion-0 emotion-1 bb-Box"
  />
  <div
    class="emotion-8 emotion-0 emotion-1 bb-Box bb-DisclosureContent"
    id="test"
  >
    <div
      class="emotion-0 emotion-1 emotion-7 bb-PageWithSidebarSidebarExpandedWrapper bb-Box"
    >
      <div
        class="emotion-0 emotion-1 emotion-4 bb-PageWithSidebarSidebar bb-Box"
      >
        <div>
          this is a sidebar
        </div>
      </div>
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-13 bb-PageWithSidebarContent bb-Box"
  >
    hello world
  </div>
</div>
`;

exports[`overrides styles.base should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  background-color: var(--bb-palette-background,#000000);
  height: 100vh;
  min-width: 320px;
  width: 320px;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  border-right: 1px solid var(--bb-palette-white800,#000000);
}

.emotion-7 {
  position: fixed;
  z-index: 999999;
  overflow-y: scroll;
}

@media screen and (max-width:1023px) {
  .emotion-7 {
    display: none;
  }
}

.emotion-13 {
  width: 100%;
  padding-left: 320px;
  padding-left: 0px;
}

@media screen and (max-width:1023px) {
  .emotion-13 {
    padding-left: 0px;
  }
}

.emotion-14 {
  min-height: 100vh;
}

.emotion-14 {
  background-color: var(--bb-palette-red,#ff0000);
}

<div
  class="emotion-14 emotion-0 emotion-1 bb-Box bb-PageWithSidebar"
>
  <div
    class="emotion-0 emotion-1 bb-Box"
  />
  <div
    class="emotion-8 emotion-0 emotion-1 bb-Box bb-DisclosureContent"
    id="test"
  >
    <div
      class="emotion-0 emotion-1 emotion-7 bb-PageWithSidebarSidebarExpandedWrapper bb-Box"
    >
      <div
        class="emotion-0 emotion-1 emotion-4 bb-PageWithSidebarSidebar bb-Box"
      >
        <div>
          this is a sidebar
        </div>
      </div>
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-13 bb-PageWithSidebarContent bb-Box"
  >
    hello world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-14 {
  min-height: 100vh;
}

.emotion-4 {
  background-color: var(--bb-palette-background,#000000);
  height: 100vh;
  min-width: 320px;
  width: 320px;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  border-right: 1px solid var(--bb-palette-white800,#000000);
}

.emotion-7 {
  position: fixed;
  z-index: 999999;
  overflow-y: scroll;
}

@media screen and (max-width:1023px) {
  .emotion-7 {
    display: none;
  }
}

.emotion-13 {
  width: 100%;
  padding-left: 320px;
  padding-left: 0px;
}

@media screen and (max-width:1023px) {
  .emotion-13 {
    padding-left: 0px;
  }
}

<div
  class="emotion-14 emotion-0 emotion-1 bb-Box bb-PageWithSidebar"
>
  <div
    class="emotion-0 emotion-1 bb-Box"
  />
  <div
    class="emotion-8 emotion-0 emotion-1 bb-Box bb-DisclosureContent"
    id="test"
  >
    <div
      class="emotion-0 emotion-1 emotion-7 bb-PageWithSidebarSidebarExpandedWrapper bb-Box"
    >
      <div
        class="emotion-0 emotion-1 emotion-4 bb-PageWithSidebarSidebar bb-Box"
      >
        <div>
          this is a sidebar
        </div>
      </div>
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-13 bb-PageWithSidebarContent bb-Box"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-14 {
  min-height: 100vh;
}

.emotion-4 {
  background-color: var(--bb-palette-background,#000000);
  height: 100vh;
  min-width: 320px;
  width: 320px;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  border-right: 1px solid var(--bb-palette-white800,#000000);
}

.emotion-7 {
  position: fixed;
  z-index: 999999;
  overflow-y: scroll;
}

@media screen and (max-width:1023px) {
  .emotion-7 {
    display: none;
  }
}

.emotion-13 {
  width: 100%;
  padding-left: 320px;
  padding-left: 0px;
}

@media screen and (max-width:1023px) {
  .emotion-13 {
    padding-left: 0px;
  }
}

.emotion-16 {
  color: var(--bb-palette-primary,#000000) !important;
}

<div
  class="emotion-14 emotion-0 emotion-16 bb-Box bb-PageWithSidebar"
>
  <div
    class="emotion-0 emotion-1 bb-Box"
  />
  <div
    class="emotion-8 emotion-0 emotion-1 bb-Box bb-DisclosureContent"
    id="test"
  >
    <div
      class="emotion-0 emotion-1 emotion-7 bb-PageWithSidebarSidebarExpandedWrapper bb-Box"
    >
      <div
        class="emotion-0 emotion-1 emotion-4 bb-PageWithSidebarSidebar bb-Box"
      >
        <div>
          this is a sidebar
        </div>
      </div>
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-13 bb-PageWithSidebarContent bb-Box"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly with sidebarPlacement 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-14 {
  min-height: 100vh;
}

.emotion-4 {
  background-color: var(--bb-palette-background,#000000);
  height: 100vh;
  min-width: 320px;
  width: 320px;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  border-left: 1px solid var(--bb-palette-white800,#000000);
}

.emotion-7 {
  position: fixed;
  z-index: 999999;
  overflow-y: scroll;
  right: 0;
}

@media screen and (max-width:1023px) {
  .emotion-7 {
    display: none;
  }
}

.emotion-13 {
  width: 100%;
  padding-right: 320px;
  padding-right: 0px;
}

@media screen and (max-width:1023px) {
  .emotion-13 {
    padding-right: 0px;
  }
}

<div
  class="emotion-14 emotion-0 emotion-1 bb-Box bb-PageWithSidebar"
>
  <div
    class="emotion-0 emotion-1 bb-Box"
  />
  <div
    class="emotion-8 emotion-0 emotion-1 bb-Box bb-DisclosureContent"
    id="test"
  >
    <div
      class="emotion-0 emotion-1 emotion-7 bb-PageWithSidebarSidebarExpandedWrapper bb-Box"
    >
      <div
        class="emotion-0 emotion-1 emotion-4 bb-PageWithSidebarSidebar bb-Box"
      >
        <div>
          this is a sidebar
        </div>
      </div>
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-13 bb-PageWithSidebarContent bb-Box"
  >
    Hello world
  </div>
</div>
`;

exports[`theming styles.base should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  background-color: var(--bb-palette-background,#000000);
  height: 100vh;
  min-width: 320px;
  width: 320px;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  border-right: 1px solid var(--bb-palette-white800,#000000);
}

.emotion-7 {
  position: fixed;
  z-index: 999999;
  overflow-y: scroll;
}

@media screen and (max-width:1023px) {
  .emotion-7 {
    display: none;
  }
}

.emotion-13 {
  width: 100%;
  padding-left: 320px;
  padding-left: 0px;
}

@media screen and (max-width:1023px) {
  .emotion-13 {
    padding-left: 0px;
  }
}

.emotion-14 {
  min-height: 100vh;
}

.emotion-14 {
  background-color: var(--bb-palette-red,#ff0000);
}

<div
  class="emotion-14 emotion-0 emotion-1 bb-Box bb-PageWithSidebar"
>
  <div
    class="emotion-0 emotion-1 bb-Box"
  />
  <div
    class="emotion-8 emotion-0 emotion-1 bb-Box bb-DisclosureContent"
    id="test"
  >
    <div
      class="emotion-0 emotion-1 emotion-7 bb-PageWithSidebarSidebarExpandedWrapper bb-Box"
    >
      <div
        class="emotion-0 emotion-1 emotion-4 bb-PageWithSidebarSidebar bb-Box"
      >
        <div>
          this is a sidebar
        </div>
      </div>
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-13 bb-PageWithSidebarContent bb-Box"
  >
    hello world
  </div>
</div>
`;

exports[`variants styles.base should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  background-color: var(--bb-palette-background,#000000);
  height: 100vh;
  min-width: 320px;
  width: 320px;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  border-right: 1px solid var(--bb-palette-white800,#000000);
}

.emotion-7 {
  position: fixed;
  z-index: 999999;
  overflow-y: scroll;
}

@media screen and (max-width:1023px) {
  .emotion-7 {
    display: none;
  }
}

.emotion-13 {
  width: 100%;
  padding-left: 320px;
  padding-left: 0px;
}

@media screen and (max-width:1023px) {
  .emotion-13 {
    padding-left: 0px;
  }
}

.emotion-14 {
  min-height: 100vh;
}

.emotion-14 {
  background-color: var(--bb-palette-red,#ff0000);
}

<div
  class="emotion-14 emotion-0 emotion-1 bb-Box bb-PageWithSidebar"
>
  <div
    class="emotion-0 emotion-1 bb-Box"
  />
  <div
    class="emotion-8 emotion-0 emotion-1 bb-Box bb-DisclosureContent"
    id="test"
  >
    <div
      class="emotion-0 emotion-1 emotion-7 bb-PageWithSidebarSidebarExpandedWrapper bb-Box"
    >
      <div
        class="emotion-0 emotion-1 emotion-4 bb-PageWithSidebarSidebar bb-Box"
      >
        <div>
          this is a sidebar
        </div>
      </div>
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-13 bb-PageWithSidebarContent bb-Box"
  >
    hello world
  </div>
</div>
`;
